### ISSUE TITLE
Refactor token swapping

### ISSUE THREAD

**Issue by ableuler** [SwissDataScienceCenter/renku-gateway#232]
The code which swapps tokens in the heade before forwarding the request should be refactored, specifically logging, code duplication, error handling and code self-documentation.

**Comment by rokroskar** [comment]
Can this be closed?

**Comment by ableuler** [comment]
> Can this be closed?

No. The part which obtains and stores token has been refactored recently, but the part acting as traffic middleware is still needs to be refactored.

**Comment by lorenzo-cavazzi** [comment]
This may also be addressed when refactoring the token code
https://github.com/SwissDataScienceCenter/renku-gateway/pull/309#discussion_r512033782
